src/analyser.cpp

```diff
@@ -0,0 +1,1509 @@
+                   && !variable->initialValue().empty()
+                   && !analyserVariable->mVariable->initialValue().empty()) {
+            auto issue = Issue::create();
+            auto trackedVariableComponent = owningComponent(analyserVariable->mVariable);
+
+            issue->setDescription("Variable '" + variable->name()
+                                  + "' in component '" + component->name()
+                                  + "' of model '" + owningModel(component)->name()
+                                  + "' and variable '" + analyserVariable->mVariable->name()
+                                  + "' in component '" + trackedVariableComponent->name()
+                                  + "' of model '" + owningModel(trackedVariableComponent)->name()
+                                  + "' are equivalent and cannot therefore both be initialised.");
+            issue->setCause(Issue::Cause::ANALYSER);
+
+            mAnalyser->addIssue(issue);
+        }
+
+        if (!analyserVariable->mVariable->initialValue().empty()
+            && !isCellMLReal(analyserVariable->mVariable->initialValue())) {
+            // The initial value is not a double, so it has to be an existing
+            // variable of constant type.
+            // Note: we always have an initialising variable. Indeed, if we were
+            //       not to have one, it would mean that the variable is
+            //       initialised using a reference to a variable that is not
+            //       defined anywhere, something that is not allowed in CellML
+            //       and will therefore be reported when we validate the model.
+
+            auto initialisingComponent = owningComponent(analyserVariable->mVariable);
+            auto initialisingVariable = initialisingComponent->variable(analyserVariable->mVariable->initialValue());
+            auto analyserInitialValueVariable = Analyser::AnalyserImpl::analyserVariable(initialisingVariable);
+
+            if (analyserInitialValueVariable->mType != AnalyserInternalVariable::Type::CONSTANT) {
+                auto issue = Issue::create();
+
+                issue->setDescription("Variable '" + variable->name()
+                                      + "' in component '" + component->name()
+                                      + "' of model '" + owningModel(component)->name()
+                                      + "' is initialised using variable '" + analyserVariable->mVariable->initialValue()
+                                      + "', but it is not a constant.");
+                issue->setCause(Issue::Cause::ANALYSER);
+
+                mAnalyser->addIssue(issue);
+            }
+        }
+    }
+
+    // Do the same for the components encapsulated by the given component.
+
+    for (size_t i = 0; i < component->componentCount(); ++i) {
+        processComponent(component->component(i));
+    }
+}
+
+void Analyser::AnalyserImpl::doEquivalentVariables(const VariablePtr &variable,
```

Comment:
Is there a clearer function name ...? I don't know what it's "doing" ?

Reply:
Well, we have a(n internal) method called `equivalentVariables()` which relies on recursion, hence I have a method called `doEquivalentVariables()`... which is used to retrieve a set of equivalent variables...?